docs/tutorials/dev_add_task.rst

```diff
@@ -0,0 +1,1405 @@
+
+        # ds = xarray.Dataset()
+        # ds['transport'] = transport
+        # ds['innerEdges'] = ('nEdges', innerEdges)
+        # write_netcdf(ds, 'transport.nc')
+
+        return innerEdges, transport
+
+
+    def _compute_barotropic_streamfunction_vertex(dsMesh, ds):
+        innerEdges, transport = _compute_transport(dsMesh, ds)
+        print('transport computed.')
+
+        nVertices = dsMesh.sizes['nVertices']
+        nTime = ds.sizes['Time']
+
+        cellsOnVertex = dsMesh.cellsOnVertex - 1
+        verticesOnEdge = dsMesh.verticesOnEdge - 1
+        isBoundaryCOV = cellsOnVertex == -1
+        boundaryVertices = numpy.logical_or(isBoundaryCOV.isel(vertexDegree=0),
+                                            isBoundaryCOV.isel(vertexDegree=1))
+        boundaryVertices = numpy.logical_or(boundaryVertices,
+                                            isBoundaryCOV.isel(vertexDegree=2))
+
+        # convert from boolean mask to indices
+        boundaryVertices = numpy.flatnonzero(boundaryVertices.values)
+
+        nBoundaryVertices = len(boundaryVertices)
+        nInnerEdges = len(innerEdges)
+
+        indices = numpy.zeros((2, 2*nInnerEdges+nBoundaryVertices), dtype=int)
+        data = numpy.zeros(2*nInnerEdges+nBoundaryVertices, dtype=float)
+
+        # The difference between the streamfunction at vertices on an inner edge
+        # should be equal to the transport
+        v0 = verticesOnEdge.isel(nEdges=innerEdges, TWO=0).values
+        v1 = verticesOnEdge.isel(nEdges=innerEdges, TWO=1).values
+
+        ind = numpy.arange(nInnerEdges)
+        indices[0, 2*ind] = ind
+        indices[1, 2*ind] = v1
+        data[2*ind] = 1.
+
+        indices[0, 2*ind+1] = ind
+        indices[1, 2*ind+1] = v0
+        data[2*ind+1] = -1.
+
+        # the streamfunction should be zero at all boundary vertices
+        ind = numpy.arange(nBoundaryVertices)
+        indices[0, 2*nInnerEdges + ind] = nInnerEdges + ind
+        indices[1, 2*nInnerEdges + ind] = boundaryVertices
+        data[2*nInnerEdges + ind] = 1.
+
+        bsfVertex = xarray.DataArray(numpy.zeros((nTime, nVertices)),
+                                     dims=('Time', 'nVertices'))
+
+        for tIndex in range(nTime):
+            rhs = numpy.zeros(nInnerEdges+nBoundaryVertices, dtype=float)
+
+            # convert to Sv
+            ind = numpy.arange(nInnerEdges)
+            rhs[ind] = 1e-6*transport.isel(Time=tIndex)
+
+            ind = numpy.arange(nBoundaryVertices)
+            rhs[nInnerEdges + ind] = 0.
+
+            M = scipy.sparse.csr_matrix((data, indices),
+                                        shape=(nInnerEdges+nBoundaryVertices,
+                                               nVertices))
+
+            solution = scipy.sparse.linalg.lsqr(M, rhs)
+
+            bsfVertex[tIndex, :] = -solution[0]
+
+        return bsfVertex
+
+
+    def _compute_barotropic_streamfunction_cell(dsMesh, bsfVertex):
```

Comment:
Here do we want to add the remapping to comparison grid?

Reply:
If you are referring to the Subpolar North Atlantic, no, let's keep this simple.